micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -769,7 +777,7 @@ public Meter remove(Meter meter) {
     public Meter removeByPreFilterId(Meter.Id preFilterId) {
         final Meter meterToRemove = preFilterIdToMeterMap.get(preFilterId);
         if (meterToRemove == null)
-            return remove(getMappedId(preFilterId));
+            return remove(mapId(preFilterId));
```

Comment:
This was the result of just blind refactor but don't we need the `isStaleId` check here?
If so, we should call the new `getMappedId` method which also does that check.

Reply:
Maybe. It'd take some time and effort to get back up to speed on this part of the code. I vaguely remember fixing some issues around removing and stale things, though. I've got a few other things distracting me at the moment. It's probably easiest to make a unit test that fails than to reason the other way around.

Reply:
TL;DR: ignore me. :)

I dug deeper and I think we should not add the stale id check.

If I add a new filter to make the id stale, the next meter registration will handle the stale ids so `removeByPreFilterId` will not see any stale ids.

If there is no meter registration between adding the filter and the `removeByPreFilterId` call, this method should carry on without checking the stale ids since nothing added a new meter since then. In fact, if it does check if the id is stale, it will not remove meters that it otherwise could have.